### DISCUSSION TITLE
Set horizontal offset position for tooltip

### DISCUSSION THREAD

**Discussion by shmoeaz** [ReactTooltip/react-tooltip#966] [Q&A] [answered]
Is there a way to offset the tooltip to the right or left?
The current offset property changes the distance away from the content element. However, I would like to offset to shift the tooltip to the right or left rather than always in the center of the content element.

**Comment by gabrieljablonski** [comment, marked as answer]
You can set the `floating-ui` middlewares manually to achieve this.
The main thing is using the `offset()` middleware with the `crossAxis` options.

[Here's an example.](https://stackblitz.com/edit/react-ts-22ya1r?file=App.tsx)
```tsx
<Tooltip
  id="my-tooltip"
  middlewares={[
    // change `crossAxis`
    offset({ mainAxis: 10, crossAxis: 50 }),
    flip(),
    shift({ padding: 5 }),
  ]}
/>
```

The other middlewares are there to keep the other tooltip positioning rules the same, since passing a value to `middlewares` overrides the defaults.

Check the [`floating-ui` docs](https://floating-ui.com/docs/middleware) for more info.

**Reply by shmoeaz** [reply]
@gabrieljablonski , thank you for responding so promptly with the solution.
